pkg/sparrow/controller_test.go

```diff
@@ -236,69 +231,57 @@ func TestChecksController_Reconcile(t *testing.T) {
 
 func TestChecksController_RegisterCheck(t *testing.T) {
 	tests := []struct {
-		name    string
-		setup   func(t *testing.T) *ChecksController
-		check   checks.Check
-		wantErr bool
+		name  string
+		setup func(t *testing.T) *ChecksController
+		check checks.Check
 	}{
 		{
 			name: "valid check",
 			setup: func(_ *testing.T) *ChecksController {
 				return NewChecksController(db.NewInMemory(), NewMetrics())
 			},
-			check:   health.NewCheck(),
-			wantErr: false,
+			check: health.NewCheck(),
 		},
 		{
 			name: "duplicate check registration",
 			setup: func(t *testing.T) *ChecksController {
 				cc := NewChecksController(db.NewInMemory(), NewMetrics())
 				check := health.NewCheck()
-				err := cc.RegisterCheck(context.Background(), check)
-				if err != nil {
-					t.Fatalf("RegisterCheck() error = %v", err)
-				}
+				cc.RegisterCheck(context.Background(), check)
 				return cc
 			},
-			check:   health.NewCheck(),
-			wantErr: false,
+			check: health.NewCheck(),
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			cc := tt.setup(t)
 
-			err := cc.RegisterCheck(context.Background(), tt.check)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("RegisterCheck() error = %v, wantErr %v", err, tt.wantErr)
-			}
+			cc.RegisterCheck(context.Background(), tt.check)
 		})
 	}
 }
 
 func TestChecksController_UnregisterCheck(t *testing.T) {
 	tests := []struct {
-		name    string
-		setup   func(t *testing.T) *ChecksController
-		check   checks.Check
-		wantErr bool
+		name  string
+		setup func(t *testing.T) *ChecksController
```

Comment:
> you can remove the `t *testing.T` argument here.

Here aswell

Reply:
I've removed the arguments from one of the tests and I've completely removed the function in the other test, as it was returning the same thing in both cases of the test suite.